### PR TITLE
New "connect" playback mode, etc.

### PR DESCRIPTION
**Pull request by ttsuchiya**
- Add new "connect" mode. To use this, the user will first need to have / create hierarchical data in CODAP, and select a higher-level collection name in the "Connect cases by..." dropdown menu. (FYI, lower-level collections equivalent to individual cases are hidden and non-selectable.) Selecting "-----" in the menu will disengage from the "connect" mode, falling back to the default "focus" mode.
- With this UI design, currently it is not possible to "connect" cases without a hierarchy in CODAP. It will always be in the default "focus" mode.
- The previous "focus" / "contrast" mode switch is now hidden. The "contrast" mode is made unavailable (for now).
- The loop toggle automatically enables clicks (at the cycle end/start).

### CODE REVIEW

File: Sonification/MicroRhythm/app.js
Comment (jsandoe) [inline Sonification/MicroRhythm/app.js]: This property doesn't seem to be used. Is it anticipating a feature?

File: Sonification/MicroRhythm/app.js
Comment (jsandoe) [inline Sonification/MicroRhythm/app.js]: Was this intended to be a temporary removal of the drag/drop feature?

**Review by jsandoe — COMMENTED** [review]
This is great work. It is much easier to track the pitch changes vs the graph. I am only not sure about the removal of drag/drop attributes.

I noticed that if I add data to the graph, the new items are not recognized in CONNECT mode. You have to toggle to FOCUS and back to CONNECT mode to get them. Is that something you knew about?

File: Sonification/MicroRhythm/app.js
Comment (ttsuchiya) [reply Sonification/MicroRhythm/app.js]: I think you're right that this has no use. I will remove it.

File: Sonification/MicroRhythm/app.js
Comment (ttsuchiya) [reply Sonification/MicroRhythm/app.js]: I thought my half-baked drag/drop feature (for CODAP attributes into plugin parameters) was not currently functional, so I disabled it. With the above uncommented, the user would see the parameter boxes visually highlighted when dragging an attribute in CODAP, but dropping into a highlighted parameter would not do anything. I thought this was rather misleading.

**Comment by ttsuchiya** [comment]
Thanks for the review, Jonathan. Good catch with the bug not updating the plugin state with CODAP data modification. I'll try handling that before merge.

File: Sonification/MicroRhythm/app.js
Comment (jsandoe) [reply Sonification/MicroRhythm/app.js]: Drag/drop is still functional for pitch and time attributes in version 0.2.8. I don't use it often. I don't know if others do. I wouldn't object to removing the feature and see if others notice...

**Comment by jsandoe** [comment]
OK, I'll wait on the merge for the fix to the updating issue.

**Comment by ttsuchiya** [comment]
Finally, patched these bugs:

- Re-enable the drag/drop for the plugin (yes it did work...).
- Adding / deleting a new case was not updating the connect mode's events.
- No sound when opening a document in the connect mode.
- Grouped-by events that do not fill to the end would hold the last note indefinitely.
- Visual playhead (tracker) was not precisely following the currently played event for both focus and connect modes. (This misalignment was noticeable with documents with just a few cases.)

If you could confirm these fixes, I think we can merge it (please feel free to do so) and announce the official "release" of v0.3.

**Review by jsandoe — APPROVED** [review]
Looks good to me

**Pull request merged**
